crates/sui-framework/packages/bridge/sources/message.move

```diff
@@ -208,13 +207,28 @@ module bridge::message {
         self.op_type
     }
 
+    fun reverse_bytes(bytes: vector<u8>): vector<u8> {
+        vector::reverse(&mut bytes);
+        bytes
+    }
+
+    fun peel_u64_be(bcs: &mut BCS): u64 {
```

Comment:
do we allow the input to be longer than 8 bytes?

Reply:
answering my own question - yes it can have trailing bytes.
nit for adding the length check before going into the loop

Reply:
the input can have more than 8 bytes, but this function don't need to care about the trailing bytes, it will always peels the first 8 bytes and return the u64 value, the remaining bytes will remain in the BCS object and consumed outside this function.